CMakeLists.txt

```diff
@@ -86,6 +86,15 @@ if(NOT MSVC)
     set_directory_properties(PROPERTIES INCLUDE_DIRECTORIES ${DL_INCLUDE_DIR})
     add_definitions(-DHAVE_DL=1)
   endif()
+else()
+  option(MSVC_USE_MP "Set to ON to build with the /MP option (Visual Studio 2005 and above)." ON)
+  mark_as_advanced(MSVC_USE_MP)
+  if(MSVC_USE_MP)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} /MP")
+  endif(MSVC_USE_MP)
+  # MSVC compilation flags suppressing compilation warnings
+  add_definitions(-D_SCL_SECURE_NO_WARNINGS)
```

Comment:
It is OK, but IMO, it's better if warnings are disabled in source code using `#pragma warning`.
So, the source code is independent and portable from a build system. 
Build systems can be changed and the same problem is back, unless it's solved at the source code level.